application/controllers/logic.py

```diff
@@ -52,81 +68,93 @@ def check_move (card, player, is_discarded, for_wonder):
         return False
     else:
         return True
-"""
 
-def process_card(card, player, is_discarded, for_wonder):
-    # checks move is valid, returns false if not
-    #if not check_move(card, player, is_discarded, for_wonder):
-    #    return False
-    if is_discarded:
-        player.money += 3
 
-        history = Cardhist(playerId=player.id, cardId=card.id, discarded=True)
-        db.session.add(history)
+def update_player(card, player):
```

Comment:
The player object is updated in this method, and then the process_card method calls the update_db method which then saves the player, saves the card playing history, and organises the next round. I guess the flow of this part isn't entirely intuitive - there's a lot of things that have to happen every time a card is played. I've tried to move most of the details into helper methods with names that you can guess their function (as the process_card method was 80 lines long before!) but if you can think of any easier ways to show what's going on, that would be useful. 👍